drivers/sensor/meas/ms5837/ms5837.c

```diff
@@ -348,34 +349,25 @@ static int ms5837_init(const struct device *dev)
 		return err;
 	}
 
-	const int type_id = (data->factory >> 5) & 0x7f;
-
-	switch (type_id) {
-	case  MS5837_02BA01:
-	case MS5837_02BA21:
-		data->comp_func = ms5837_compensate_02;
-		break;
-	case MS5837_30BA26:
-		data->comp_func = ms5837_compensate_30;
-		break;
-	default:
-		LOG_WRN(" unrecognized type: '%2x', defaulting to MS5837-30", type_id);
-		data->comp_func = ms5837_compensate_30;
-		break;
-	}
-
 	return 0;
 }
 
-#define MS5837_DEFINE(inst)								\
-	static struct ms5837_data ms5837_data_##inst;					\
-											\
-	static const struct ms5837_config ms5837_config_##inst = {			\
-		.i2c = I2C_DT_SPEC_INST_GET(inst),					\
-	};										\
-											\
-	SENSOR_DEVICE_DT_INST_DEFINE(inst, ms5837_init, NULL,				\
-			      &ms5837_data_##inst, &ms5837_config_##inst, POST_KERNEL,	\
-			      CONFIG_SENSOR_INIT_PRIORITY, &ms5837_api_funcs);		\
-
-DT_INST_FOREACH_STATUS_OKAY(MS5837_DEFINE)
+#define MS5837_DEFINE(inst, type)                                                                  \
+	static struct ms5837_data ms5837_data_##inst;                                              \
+                                                                                                   \
+	static const struct ms5837_config ms5837_config_##inst = {                                 \
```

Comment:
These symbols need to include `type` in case there are instances of both compatibles in the devicetree.
```suggestion
	static struct ms5837_data ms5837_##type##_data_##inst;                                              \
                                                                                                   \
	static const struct ms5837_config ms5837_##type##_config_##inst = {                                 \
```

Reply:
I added your suggestions and also adjusted the SENSOR_DEVICE_DT_INST_DEFINE call to the new names that include the type as well. Thank you.